### PR TITLE
Allow plural nouns when running 'pachctl list' commands

### LINKED ISSUE #5183: Support plurals in pachctl list & friends
My instinct, as a long time `kubectl` user, is to type `pachctl list pipelines`
However, the less grammatical `pachctl list pipeline` is the only valid spelling.
Add support for plurals in commands where it makes sense.

### PR DESCRIPTION
**Pull request by Tryneus**
Fixes #5183.

I looked through the existing commands and pluralized the ones that made sense - mostly just `list` commands and `flush jobs` and `restart datums` - none of the other commands take multiple targets.  The pluralized nouns are hidden from help to avoid polluting it.

### CODE REVIEW

**Comment by jdoliner** [comment]
Being more conservative seems like a reasonable approach. I think you should probably add `flush commits`, it returns multiple commits, and can accept multiple commits as arguments, and is pretty analogous to `flush job` which this pluralizes. Also `subscribe commit` is more or less the same semantics as `list commit` except it's streaming, so that seems like a good one to add as well.
 Other candidates that I don't feel too strongly about either way:

- All file operations can operate on multiple files.
- `create pipeline` can create multiple pipelines

**Comment by Tryneus** [comment]
Good points, added those.  `inspect file` doesn't seem to operate on multiple files or globs, so I've left that one out.

File: src/server/pkg/cmdutil/cobra.go [hunk 1]
```diff
@@ -311,6 +311,18 @@ func (r *RepeatedStringArg) Type() string {
 	return "[]string"
 }
 
+// Hide marks all runnable subcommands as hidden and returns it
+// for method chaining.
+func Hide(cmd *cobra.Command) *cobra.Command {
```
Comment (msteffen) [inline src/server/pkg/cmdutil/cobra.go:316]: Just checking my understanding here, this is so if someone types `pachctl list` they don't get a huge autocomplete of like `repo, repos, job, jobs, file, files, ...`?

**Review by msteffen — APPROVED** [review]
Sorry for taking so long, I totally missed that this PR came in. I have a new chrome extension that will hopefully help me do a better job staying on top of PRs like this.

PR itself LGTM (one question just out of curiosity)

**Comment by jdoliner** [comment]
I just poked CI here and it passed. @Tryneus is this good to merge?

**Pull request closed without merging**